### DISCUSSION TITLE
How to update sherlock

### DISCUSSION THREAD

**Discussion by Ch4054u** [sherlock-project/sherlock#1112] [Q&A]
I do not think that my sherlock I have, has all the sites that i am seeing should be there... seems like sometimes it just dont work.

**Comment by Ch4054u** [comment]
How do u update all. I don’t think it has but just a few

**Reply by DavidMBeetle** [reply]
This only really applies to home brew but 2 commands work:

brew update sherlock

brew upgrade sherlock

brew update sherlock seems to update it their and then and brew upgrade sherlock seems to enable auto updating of some kind.


<img width="2982" height="344" alt="Terminal screenshot" src="https://github.com/user-attachments/assets/94c44b02-d60a-41aa-8e75-c31609d61f4f" />

**Reply by SethFalco** [reply]
Just a heads-up for future reference!

The two commands you did were identical in your case, you did not have to do `brew upgrade sherlock` after doing `brew update sherlock`. The `update` command was automatically redirected to `upgrade`.

`brew update` is not meant to receive a package name. With the intended usage, it just updates homebrew itself, and then **checks** for software updates _without actually upgrading your software_. Homebrew determined you made a likely mistake and assumed you meant to do `brew upgrade sherlock` instead.

`brew upgrade [package]` is then used to actually perform software updates.

TL;DR:

✅ `brew update`: check for updates.
✅ `brew upgrade [package]`: perform the update.
❌ `brew update [package]`: not a documented command, but brew redirects it to `brew upgrade [package]` because it's probably what you mean.

**Comment by Ch4054u** [comment]
Just start by giving me some lol I’m new here so I’ll just start to update
this one by one if I have too

**Comment by Ch4054u** [comment]
I developed programs n games but new too it. Trying to learn Linux
Can u help me

**Comment by Ch4054u** [comment]
For one, to get my Sherlock to work right. It seems like it’s the very
first version, like it don’t have many sites

**Comment by Ch4054u** [comment]
What kinda phone u have? I have iPhone… video call or something so I can
show u what it is doing

**Comment by Ch4054u** [comment]
Can u help

**Comment by MrHaanMan** [comment]
Try using the **--print-all** argument . By default it supresses the sites where the username is not found. So looks like its not doing much. 
![image](https://user-images.githubusercontent.com/58526807/144034953-3eae9809-fb83-407e-ba0f-e53aa46610d4.png)

**Comment by SethFalco** [comment]
Assuming you followed the installation instructions on the README:
https://github.com/sherlock-project/sherlock#installation=

You can get the up-to-date version by doing this command in that directory:
```sh
git pull
```

> seems like sometimes it just dont work.

It helps if elaborate on "just dont work". Does it throw an error, does it print anything at all?
Chances are your issue is resolved by: https://github.com/sherlock-project/sherlock/discussions/1112#discussioncomment-1721551

**Comment by ayuuXploits** [comment]
# Option 1: If installed via Git (repo clone)
cd sherlock && git pull origin master && python3 -m pip install -r requirements.txt

# Option 2: If installed via pip / pipx
python3 -m pip install --upgrade sherlock-project
# or: pipx upgrade sherlock-project

# Option 3: If installed via Docker
docker pull sherlock/sherlock

# Check version to confirm update
sherlock --version
